mikeio1d/cross_sections/cross_section.py

```diff
@@ -0,0 +1,853 @@
+        pls.Option = ProcessingOption.UserDefinedLevels
+        pls.UserDefLevels = tuple(levels)
+        self._m1d_cross_section.BaseCrossSection.CalculateProcessedData()
+
+    @property
+    def processing_levels_method(self) -> ProcessLevelsMethod:
+        """
+        The method used to generate processing levels.
+
+        Parameters
+        ----------
+        processing_level_method: int | ProcessLevelsMethod
+            0 - Automatic
+            1 - Equidistant
+            2 - User defined
+
+        Returns
+        -------
+        ProcessLevelsMethod
+            The method used to generate processing levels.
+        """
+        return ProcessLevelsMethod(
+            int(self._m1d_cross_section.BaseCrossSection.ProcessingLevelsSpecs.Option)
+        )
+
+    @processing_levels_method.setter
+    def processing_levels_method(self, processing_level_method: int | ProcessLevelsMethod):
+        self._m1d_cross_section.BaseCrossSection.ProcessingLevelsSpecs.Option = ProcessingOption(
+            int(processing_level_method)
+        )
+        self._m1d_cross_section.BaseCrossSection.CalculateProcessedData()
+
+    @property
+    def processed_allow_recompute(self) -> bool:
+        """
+        Whether the processed data can be recomputed (e.g. if the raw data has changed).
+
+        Setting this to False will freeze the processed data values in their current state.
+
+        Default is True.
+        """
+        return not self._m1d_cross_section.BaseCrossSection.ProcessedDataProtected
+
+    @processed_allow_recompute.setter
+    def processed_allow_recompute(self, value: bool):
+        self._m1d_cross_section.BaseCrossSection.ProcessedDataProtected = not value
+
+    def recompute_processed(self):
+        """
+        Recompute the processed data.
+
+        In most cases this is not necessary as it will be done automatically when the raw data changes.
+        If processed_allow_recompute is set to False, then this will do nothing.
+        """
+        self._m1d_cross_section.BaseCrossSection.CalculateProcessedData()
+
+    def _calculate_conveyance_factor(
+        self, resistance: Tuple[float], flow_area: Tuple[float], radius: Tuple[float]
+    ) -> Tuple[float]:
+        """
+        Calculate the conveyance factor from resistance, flow area and radius.
+
+        Note that this is not the true conveyance factor since resistance could be relative. However,
+        this is the same calculation provided by MIKE+ cross section editor. Its main purpose is for checking
+        that conveyance values will be monotonically increasing. For more info, see
+        the MIKE+ documentation:
+        https://doc.mikepoweredbydhi.help/webhelp/2024/MIKEPlus/MIKEPlus/RiverNetwork_HydrModel/Processed_Data.htm#XREF_52990_Processed_Data:~:text=Note%3A%20The%20conveyance,from%20the%20simulations.
+        """
+        return tuple(r * A * R ** (2.0 / 3) for r, A, R in zip(resistance, flow_area, radius))
```

Comment:
I think these type of calculations should be avoided. The reason is that quite quickly we will start implementing the computations inconsistently in different places. Also such conveyance formula holds only for Manning formulation. I thought processed data already had these type of factors. Or?

Reply:
Upon trying to use this method, it doesn't support relative resistances. I believe MIKE+ still calculates this fake conveyance even with relative resistances. It seems it's useful for modelers to have, per the comment, so I am reluctant to get rid of such an indicator. Any other suggestions?